actioncable/lib/action_cable/connection/web_socket.rb

```diff
@@ -34,8 +32,21 @@ def rack_response
         websocket.rack_response
       end
 
+      # by order of preference, the first supported client will be chosen
+      CLIENT_SOCKET_CANDIDATES = [ClientRackSocket, ClientFayeSocket]
+
+      def self.establish_connection(env, event_target, event_loop, protocols)
+        @client_socket_klass ||= client_socket_selector(env)
+        @client_socket_klass&.attempt(env, event_target, event_loop, protocols)
+      end
+
       private
         attr_reader :websocket
+
+        def self.client_socket_selector(env)
+          CLIENT_SOCKET_CANDIDATES.each { |klass| return klass if klass.accept?(env) }
+          nil
+        end
```

Comment:
Use `.detect` rather than returning within an `.each` block.

Reply:
Thanks! I never knew about `.detect` - I thought you were using shorthand / pseudo code 👏🏻👍🏻